indicators/views.py

```diff
@@ -2104,37 +2106,124 @@ def get_context_data(self, **kwargs):
-        context['active'] = ['indicators']
-        return context
+        if objective:
+            return JsonResponse(model_to_dict(objective))
+        else:
+            return JsonResponse(dict(error='Failed'))
 
 
-def objective_delete(request, pk):
+class ObjectiveUpdate(GView):
     """
-    Delete strategic objective
-    :param request:
-    :param pk:
-    :return:
+    View to Update Objective and return Json response
+    """
+    def put(self, request, *args, **kwargs):
+        objective_id = int(self.kwargs.get('id'))
+        data = json.loads(request.body.decode('utf-8'))
+        objective_name = data.get('name')
+        objective_description = data.get('description')
+        objective_parent = int(data.get('parent')) if data.get('parent') else None
+        objective_program = int(data.get('program'))
+        
+        objective = Objective.objects.get(
+            id=objective_id
+        )
+        objective.name = objective_name
+        objective.description = objective_description
+        objective.program_id = objective_program
+        objective.parent_id = objective_parent
+        objective.save()
+        if objective:
+            return JsonResponse(model_to_dict(objective))
+        else:
+            return JsonResponse(dict(error='Failed'))
+
+
+class ObjectiveDelete(GView):
+    """
+    View to Delete Objective and return Json response
     """
-    objective = Objective.objects.get(pk=int(pk))
-    objective.delete()
-    return redirect('/indicators/objectives')
+    def delete(self, request, *args, **kwargs):
+        objective_id = int(self.kwargs.get('id'))
+        objective = Objective.objects.get(
+            id=int(objective_id)
+        )
+        objective.delete()
+
+        try:
+            Objective.objects.get(id=int(objective_id))
+            return JsonResponse(dict(error='Failed'))
+
+        except Objective.DoesNotExist:
+            return JsonResponse(dict(success=True))
+
+
+# class ObjectiveUpdateView(UpdateView):
+#     model = Objective
+#     template_name_suffix = '_update_form'
+#     success_url = '/indicators/objectives'
+#     form_class = ObjectiveForm
+
+#     # add the request to the kwargs
+#     def get_form_kwargs(self):
+#         kwargs = super(ObjectiveUpdateView, self).get_form_kwargs()
+#         kwargs['request'] = self.request
+#         kwargs['current_objective'] = self.get_object()
+#         return kwargs
+
+#     def get_context_data(self, **kwargs):
+#         context = super(ObjectiveUpdateView, self).get_context_data(**kwargs)
+#         context['current_objective'] = self.get_object()
+#         context['active'] = ['indicators']
+#         return context
+
+
+# def objective_delete(request, pk):
+#     """
+#     Delete strategic objective
+#     :param request:
+#     :param pk:
+#     :return:
+#     """
+#     objective = Objective.objects.get(pk=int(pk))
+#     objective.delete()
+#     return redirect('/indicators/objectives')
```

Comment:
Can we please replace it with DRF based view?
something like https://github.com/hikaya-io/activity/pull/253/files#diff-6f3970f06e008619ad6ddaf489027baeR2311-R2315